libraries/lib-ffmpeg-support/wrappers/AVCodecContextWrapper.cpp

```diff
@@ -79,24 +79,39 @@ AVCodecContextWrapper::DecodeAudioPacket(const AVPacketWrapper* packet)
       std::unique_ptr<AVPacketWrapper> packetCopy =
          packet ? packet->Clone() : mFFmpeg->CreateAVPacketWrapper();
 
-      int packetSizeLeft = packet ? packetCopy->GetSize() : 0;
-
-      if (!packetSizeLeft)
-         packetSizeLeft = 1;
+      /*
+       "Flushing is done by calling this function [avcodec_decode_audio4]
+       with packets with avpkt->data set to NULL and avpkt->size set to 0
+       until it stops returning samples."
+       (That implies, not necessarily just one loop pass to flush)
+       */
+      bool flushing = packet
```

Comment:
This looks like a more correct implementation. However, FFmpeg itself doesn't bother about flushing at all. We did it only once (and I was trying to keep as much of the original behavior as possible). 

I think codecs with AV_CODEC_CAP_DELAY are not common at all and the best implementation should check for it. Very quick search and I found only APE decoder to set this flag, but I think we do not support APE.